CODEOWNERS

```diff
@@ -0,0 +1,101 @@
+# https://help.github.com/articles/about-codeowners/
+
+# This file is called CODEOWNERS because it is a magic file for GitHub to
+# automatically suggest reviewers. In this project's case, the names below
+# should be thought of as code reviewers rather than owners. Regular
+# contributors are free to add their names to specific directories or files
+# provided that they are willing to provide a review when automatically
+# assigned.
+
+# Information on the decision making process for pull requests can be found
+# in the "Decision Making Process" section of the CONTRIBUTING.md doc.
+
+
+# Maintainers
+#   @laanwj
+#   @sipa
+#   @fanquake
+#   @jonasschnelli
+#   @marcofalke
+#   @meshcollider
+
+
+# Build system
+/configure.ac                       @dongcarl
+/build-aux/                         @dongcarl
+/depends/                           @dongcarl
+*.am                                @dongcarl
+Makefile.*                          @dongcarl
+
+# Test framework
+/src/test/                          @marcofalke
+/test/                              @marcofalke
+/ci/                                @marcofalke
+
+# Fuzzing
+/src/test/fuzz/                     @practicalswift
+/doc/fuzzing.md                     @practicalswift
+
+# GUI
+/src/qt/                            @jonasschnelli
+
+# Wallet
+/src/wallet/                        @meshcollider
+
+# Descriptors
+*descriptor*                        @achow101 @sipa
+
+# MSVC build system
+/build_msvc/                        @sipsorcery
+
+# scripts/linter
+*.sh                                @practicalswift
+/test/lint/                         @practicalswift
+/ci/lint/                           @practicalswift
+
+# Bech32
+/src/bech32.*                       @sipa
+/src/bench/bech32.*                 @sipa
+
+# PSBT
+/src/psbt*                          @achow101
+/src/node/psbt*                     @achow101
+/doc/psbt.md                        @achow101
+
+# Consensus
+/src/consensus/                     @sipa
+/src/consensus/tx_verify.*          @jtimon
+/src/consensus/validation.h         @TheBlueMatt
+/src/consensus/tx_check.*           @jnewbery
+
+# Docs
+/doc/Doxyfile.in                     @fanquake
+/doc/JSON-RPC-interface.md           @harding
+/doc/REST-interface.md               @jonasschnelli
+/doc/benchmarking.md                 @ariard
+/doc/bitcoin-conf.md                 @hebasto
+/doc/build-freebsd.md                @fanquake
+/doc/build-netbsd.md                 @fanquake
+/doc/build-openbsd.md                @laanwj
+/doc/build-osx.md                    @fanquake
+/doc/build-unix.md                   @laanwj
+/doc/build-windows.md                @laanwj
+/doc/dependencies.md                 @fanquake
+/doc/developer-notes.md              @laanwj
+/doc/files.md                        @hebasto
+/doc/gitian-building.md              @laanwj
+/doc/productivity.md                 @dongcarl
+/doc/rapidcheck.md                   @jonatack
```

Comment:
The rapidcheck documentation was recently removed along with rapidcheck itself.

Reply:
Ah, thanks. This section currently serves as an example of what is possible, but your comment also illustrates the point in the PR description that the downside of a detailed CODEOWNERS file is that it will require more attention and updates.

Reply:
> your comment also illustrates the point in the PR description that the downside of a detailed CODEOWNERS file is that it will require more attention and updates.

Agreed. It may be good for someone to be nominated/dedicated to this.